File: Cristian/congruencia.py
print ("-------------------------------------")
print ("- Metodo de Congruencia Fundamental -")
print ("-------------------------------------")
# print (" 	 v3 = (a*v2 + c*v1) % m 		")
print ("   v(n+1) = (a * v(n) + c) % m ")
print ("-------------------------------------")

v1 = int(input('Valor de la semilla (valor inicial): '))
m = int(input('Valor de m: '))
a = int(input('Valor de a: '))
c = int(input('Valor de c: '))
n = int(input('Cantidad de Iteraciones: '))

if not((v1>0) and (m>0) and (a>0) and (c>0)):
	print ("Error: Todos los valores deben ser mayores a 0")
elif not(m>a):
	print ("Error: m debe mayor que a")
else:
	vector_resultados = []
	v2 = (a * v1 + c) % m
	vector_resultados.append(v2)
	for x in range(1, n):
		v3 = (a * v2 + c) % m
		vector_resultados.append(v3)
		v1=v2
		v2=v3
	print ('Resultados:')
	print (vector_resultados)

File: Cristian/congruencia.js

/**
 * 
 * @param {*} v1 
 * @param {*} v2 
 * @param {*} k 
 * @param {*} m 
 * @param {*} a 
 * @param {*} c 
 * @param {*} i 
 */



function congruencia(v1, m, a, c, i) {
    if (!(v1 > 0 && m > 0 && a > 0 && c > 0)) {
        console.error("Error: Todos los valores deben ser mayores a 0");
        return;
    } else if (!(m > a)) {
        console.error("Error: m debe ser mayor que a.");
        return;
    } else {
        const resultado = [];
        let v2 = (a * v1 + c) % m;
        resultado.push(v2);
        for (let x = 1; x < i; x++) {
            v3 = (a * v2 + c) % m;
            resultado.push(v3);
            v1 = v2;
            v2 = v3;
        }
        return resultado;
    }

    //console.log("Resultados: [", resultado, "]");
}

const v1 = 117;
const m = 1000;
const a = 4;
const c = 8;
const i = 15;

console.log("conf: ", congruencia(v1, m, a, c, i));
